rllib/algorithms/algorithm_config.py

```diff
@@ -2430,6 +2439,12 @@ def offline_data(
         """
         if input_ is not NotProvided:
             self.input_ = input_
+        if input_read_method is not NotProvided:
+            self.input_read_method = input_read_method
+        if input_read_method_kwargs is not NotProvided:
+            self.input_read_method_kwargs = input_read_method_kwargs
+        else:
```

Comment:
we shouldn't set defaults here. Instead, set this property to its default value `{}` already in the constructor.